src/modules/simulation/gz_plugins/moving_platform_controller/MovingPlatformController.hpp

```diff
@@ -0,0 +1,123 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2025 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *	notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *	notice, this list of conditions and the following disclaimer in
+ *	the documentation and/or other materials provided with the
+ *	distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *	used to endorse or promote products derived from this software
+ *	without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#pragma once
+
+#include <gz/sim/Util.hh>
+#include <gz/sim/Model.hh>
+#include <gz/sim/System.hh>
+#include "gz/sim/components/Pose.hh"
+#include <gz/sim/components/Model.hh>
+#include <gz/sim/EntityComponentManager.hh>
+
+#include <gz/transport/Node.hh>
+#include <gz/plugin/Register.hh>
+#include <gz/msgs/Utility.hh>
+#include <gz/msgs/twist.pb.h>
+
+#include <gz/math.hh>
+#include <gz/math/Rand.hh>
+#include <gz/math/Pose3.hh>
+
+#include <iostream>
+
+
+/*
+ *  To use this plugin, add the following to your model.sdf:
+ *
+ *      <?xml version="1.0" encoding="UTF-8"?>
+ *      <sdf version="1.9">
+ *        <model name="flat_platform">
+ *
+ *          <!-- the rest of your model -->
+ *
+ *          <plugin
+ *            filename="gz-sim-velocity-control-system"
+ *            name="gz::sim::systems::VelocityControl">
+ *          </plugin>
+ *
+ *          <plugin
+ *            filename="libMovingPlatformController.so"
+ *            name="custom::MovingPlatformController">
+ *          </plugin>
+ *
+ *        </model>
+ *      </sdf>
+ *
+ */
```

Comment:
maybe this comment and the mention of the env var can go in a README.md in this plugin dir? I don't know... might be helpful to have info consolidated